### PR TITLE
MODRS-6 - Retrieval queue records API

### PR DESCRIPTION
**Pull request by khandramai**
[MODRS-6](https://issues.folio.org/browse/MODRS-6) - Requesting (Retrieval) a Remote Storage Item

## Purpose
API for retrievals should be implemented for interaction with edge-dematic.

## Approach
GET retrievals, PUT retrievals (to set retrieved/created) endpoints are implemented

## Pre-Merge Checklist:
 Before merging this PR, please go through the following list and take appropriate actions.

 - Does this PR meet or exceed the expected quality standards?
   - [x] Code coverage on new code is 80% or greater
   - [x] Duplications on new code is 3% or less
   - [x] There are no major code smells or security issues
 - Does this introduce breaking changes?
   - [ ] Were any API paths or methods changed, added or removed?
   - [ ] Were there any schema changes?
   - [ ] Did any of the interface versions change?
   - [ ] Were permissions changed, added, or removed?
   - [ ] Are there new interface dependencies?
   - [x] There are no breaking changes in this PR.

 If there are breaking changes, please **STOP** and consider the following:

 - What other modules will these changes impact?
 - Do JIRAs exist to update the impacted modules?
   - [ ] If not, please create them
   - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
   - [ ] Do they have all they appropriate links to blocked/related issues?
 - Are the JIRAs under active development?
   - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
 - Do PRs exist for these changes?
   - [ ] If so, have they been approved?

 Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

 While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by khandramai** [comment]
This PR can be merge before [#36](https://github.com/folio-org/mod-remote-storage/pull/36). @VitaliiZahoruiko, please pay attention and save RetrievalQueueRecord content.

**Review by alekGbuz — APPROVED** [review]

File: src/test/java/org/folio/rs/service/RetrievalQueueServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    ResponseEntity<RetrievalQueues> responseEntity = get(formattedRetrievalUrl + "?limit=1", RetrievalQueues.class);
+    assertThat(Objects.requireNonNull(responseEntity.getBody())
+      .getTotalRecords(), equalTo(2));
+    assertThat(Objects.requireNonNull(responseEntity.getBody())
+      .getRetrievals()
+      .size(), equalTo(1));
+  }
+
+  @Test
+  void shouldSetAccessionedById() {
```
Comment (siarhei-charniak) [inline src/test/java/org/folio/rs/service/RetrievalQueueServiceTest.java:117]: Please rename Accessioned -> Retrieved

File: src/test/java/org/folio/rs/service/RetrievalQueueServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    UUID id = UUID.randomUUID();
+    retrievalQueueRepository.save(buildRetrievalQueueRecord(id));
+
+    put(formattedRetrievalUrl + "/id/" + id, null);
+
+    var actualAccessionQueueRecord = retrievalQueueRepository.findAll()
```
Comment (siarhei-charniak) [inline src/test/java/org/folio/rs/service/RetrievalQueueServiceTest.java:123]: Please rename Accession -> Retrieval

File: src/test/java/org/folio/rs/service/RetrievalQueueServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+      .get(0);
+    assertThat(actualAccessionQueueRecord.getRetrievedDateTime(), notNullValue());
+  }
+
+  @Test
+  void shouldSetAccessionedByBarcode() {
```
Comment (siarhei-charniak) [inline src/test/java/org/folio/rs/service/RetrievalQueueServiceTest.java:129]: Please rename Accessioned -> Retrieved

File: src/test/java/org/folio/rs/service/RetrievalQueueServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+    retrievalQueueRepository.save(buildRetrievalQueueRecord(UUID.randomUUID()));
+
+    put(formattedRetrievalUrl + "/barcode/" + BARCODE, null);
+
+    var actualAccessionQueueRecord = retrievalQueueRepository.findAll()
```
Comment (siarhei-charniak) [inline src/test/java/org/folio/rs/service/RetrievalQueueServiceTest.java:134]: Please rename Accession -> Retrieval

File: src/test/java/org/folio/rs/service/RetrievalQueueServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+    retrievalQueueRepository.save(buildRetrievalQueueRecord(UUID.randomUUID()));
+    String url = formattedRetrievalUrl + "/id/" + UUID.randomUUID();
+
+    HttpClientErrorException exception = assertThrows(HttpClientErrorException.class, () -> put(url, null));
+
+    assertThat(exception.getStatusCode(), Matchers.is(HttpStatus.NOT_FOUND));
+  }
+
+  @Test
+  void shouldThrowNotFoundExceptionWhenAccessionQueueIsAlreadyAccessioned() {
```
Comment (siarhei-charniak) [inline src/test/java/org/folio/rs/service/RetrievalQueueServiceTest.java:151]: Please rename Accessioned -> Retrieved

**Review by siarhei-charniak — APPROVED** [review]
@khandramai this PR looks good. Please fix comments regarding method and variable names.

**Pull request merged**